Tools Used:
- Google Colab
- Python
- Gemini API
- ChatGPT

In [ ]:
!pip install -q -U google-genai

In [ ]:
from google import genai
from google.colab import userdata

api_key = userdata.get("GEMINI_API_KEY")

client = genai.Client(api_key=api_key)

print("Gemini API connected successfully!")

Gemini API connected successfully!


In [ ]:
customer_message = """
My phone charger arrived late, and the cable is not charging my phone.
I would like a replacement.
"""

print(customer_message)


My phone charger arrived late, and the cable is not charging my phone.
I would like a replacement.



In [ ]:
prompt1 = f"""
You are a customer support assistant.

Classify the following customer complaint. Identify:
1. The main issue
2. The complaint category
3. The customer's requested solution

Use clear labels. Do not invent information.

Customer complaint:
{customer_message}
"""

response1 = client.models.generate_content(
    model="gemini-3.8-flash",
    contents=prompt1
)

classification = response1.text

print("STEP 1 - CLASSIFICATION")
print(classification)

STEP 1 - CLASSIFICATION
**1. Main Issue:** The phone charger arrived late, and the cable does not charge the phone.

**2. Complaint Category:** Delivery Delay / Defective Product (Hardware Issue)

**3. Requested Solution:** A replacement.


In [ ]:
prompt2 = f"""
You are a customer support information-gathering assistant.

Based on the customer complaint and classification below, identify the missing
information needed to process the customer's replacement request.

List only the important missing information.
Do not invent facts.
Return the answer as a numbered list.

Customer complaint:
{customer_message}

Classification from Step 1:
{classification}
"""

response2 = client.models.generate_content(
    model="gemini-3.8-flash",
    contents=prompt2
)

missing_information = response2.text

print("STEP 2 - MISSING INFORMATION")
print(missing_information)

STEP 2 - MISSING INFORMATION
Based on the customer's complaint, the following important information is missing to process the replacement request:

1. **Order number** (or proof of purchase/account details) to locate the transaction.
2. **Product details** (specific charger/cable model or phone model) to ensure the correct replacement is sent.
3. **Shipping address** to confirm where the replacement should be delivered.


In [ ]:
prompt3 = f"""
You are a professional customer support representative.

Recommend the best solution using the information below.

Requirements:
- Use a polite and professional tone.
- Do not promise a replacement until the information is verified.
- Explain the next action clearly.
- Keep the response under 120 words.

Customer complaint:
{customer_message}

Classification:
{classification}

Missing information:
{missing_information}
"""

response3 = client.models.generate_content(
    model="gemini-3.8-flash",
    contents=prompt3
)

recommended_solution = response3.text

print("STEP 3 - RECOMMENDED SOLUTION")
print(recommended_solution)

ClientError: 429 RESOURCE_EXHAUSTED. {'error': {'code': 429, 'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, head to: https://ai.google.dev/gemini-api/docs/rate-limits. To monitor your current usage, head to: https://ai.dev/rate-limit. \n* Quota exceeded for metric: generativelanguage.googleapis.com/generate_content_free_tier_requests, limit: 5, model: gemini-3.8-flash\nPlease retry in 16.738405502s.', 'status': 'RESOURCE_EXHAUSTED', 'details': [{'@type': 'type.googleapis.com/google.rpc.Help', 'links': [{'description': 'Learn more about Gemini API quotas', 'url': 'https://ai.google.dev/gemini-api/docs/rate-limits'}]}, {'@type': 'type.googleapis.com/google.rpc.QuotaFailure', 'violations': [{'quotaMetric': 'generativelanguage.googleapis.com/generate_content_free_tier_requests', 'quotaId': 'GenerateRequestsPerMinutePerProjectPerModel-FreeTier', 'quotaDimensions': {'location': 'global', 'model': 'gemini-3.8-flash'}, 'quotaValue': '5'}]}, {'@type': 'type.googleapis.com/google.rpc.RetryInfo', 'retryDelay': '16s'}]}}

ChatGPT Output:

The best solution is to verify the customer’s order number, product details,
and shipping address. Customer service should confirm that the cable is
defective and then process a replacement. The late delivery should also be
reviewed, and the customer should receive an updated delivery estimate.


In [ ]:
prompt4 = f"""
You are a customer support escalation assistant.

Decide whether this complaint should be escalated to a human support agent.

Escalate if:
- There is a safety concern.
- There is a serious billing issue.
- The customer has already tried and failed to receive help.
- The issue requires a policy exception or human investigation.

Return:
Decision: ESCALATE or DO NOT ESCALATE
Reason: One sentence explaining the decision.
Next action: One sentence explaining what should happen next.

Customer complaint:
{customer_message}

Classification:
{classification}

Missing information:
{missing_information}

Recommended solution:
{recommended_solution}
"""

response4 = client.models.generate_content(
    model="gemini-3.8-flash",
    contents=prompt4
)

escalation_decision = response4.text

print("STEP 4 - ESCALATION DECISION")
print(escalation_decision)

ChatGPT Output:

Decision: DO NOT ESCALATE

Reason: The customer reported a late delivery and defective cable, but there
is no reported safety concern, serious billing issue, or previous failed
resolution.

Next action: Collect the missing information, verify the defective cable, and
process the replacement.

Testing and Iteration:

The first two prompts ran successfully. Step 3 returned a 429 quota error
because the Gemini API quota was exceeded. I completed the remaining steps
using manual ChatGPT prompting and included the successful outputs.

EX)

Before:
Identify missing information.

After:
Based on the customer complaint and classification, identify only the
information needed to process the replacement request. Return the answer as a
numbered list and do not invent any facts.